unicorn_mode/build_unicorn_support.py

```diff
@@ -0,0 +1,154 @@
+
+def run_cmd(cmd: str, cwd: Path = None, quiet: bool = False):
+    if not cwd:
+        cwd = Path(__file__).parent
+    if quiet:
+        print(f"[*] Runing queitly: \"{cmd}\" under workding directory {cwd}")
+    else:
+        print(f"[*] Runing: \"{cmd}\" under workding directory {cwd}")
+    if quiet:
+        try:
+            out = subprocess.check_output(cmd, shell=True, stderr=subprocess.PIPE, cwd=cwd)
+        except subprocess.CalledProcessError as e:
+            print(f"[!] Failed with:\n{e.stdout}\n{e.stderr}")
+            raise e
+        return out
+    else:
+        subprocess.check_call(cmd, shell=True, cwd=cwd)
+        return None
+
+
+def detect_from_env_or_file(target: str):
+    if target in os.environ:
+        return os.environ[target]
+    elif (cwd / target).exists():
+        with open(cwd / target, mode="r") as f:
+            return f.read().strip()
+    else:
+        return None
+    
+if sys.platform == 'win32':
+    print("[!] Unicornafl does not support Windows so far (no fsrv support).")
+    exit(1)
+    
+cwd = Path(__file__).parent
+
+libs_path = cwd / "lib"
+include_path = cwd / "include"
+unicornafl_path = cwd / "unicornafl"
+
+if libs_path.exists():
+    print("[!] Cleaning previous artifacts...")
+    shutil.rmtree(libs_path)
+if include_path.exists():
+    print("[!] Cleaning previous headers...")
+    shutil.rmtree(include_path)
+
+if not (cwd.parent / "afl-showmap").exists():
+    print("[!] Please compile AFL++ firstly.")
+    exit(1)
+
+if not shutil.which("cargo"):
+    print("[!] No cargo, please install Rust in advance.")
+    print("[!] TLDR: `curl https://sh.rustup.rs -sSf | sh -s -- -y`")
+    exit(1)
+
+unicornafl_version = detect_from_env_or_file("UNICORNAFL_VERSION")
+if not unicornafl_version:
+    print("[!] No valid UNICORNAFL_VERSION found")
+    exit(1)
+
+try:
+    run_cmd("git status", cwd)
+except subprocess.CalledProcessError:
+    print(f"[!] Unicornafl is not a submodule, do a separate clone...")
+    run_cmd("rm -rf unicornafl && git clone https://github.com/AFLplusplus/unicornafl", cwd)
+
+if not (unicornafl_path / ".git").exists():
+    print(f"[!] Submodule not existing, will do a checkout firstly")
+    run_cmd("git submodule update --init --recursive", cwd)
+
+print(f"[*] We will checkout unicornafl {unicornafl_version}")
+run_cmd(f"git fetch --all && git checkout {unicornafl_version}", unicornafl_path)
+
+print(f"[*] Now building unicornafl python bindings")
+venv = in_venv()
+if not venv:
+    print(f"[!] A python venv is highly recommended! We will try to add --user for you.")
+
+if not shutil.which("maturin"):
+    print(f"[!] No maturin, will install maturin firstly")
+    if not venv:
+        run_cmd("pip install --user maturin")
+    else:
+        run_cmd("pip install maturin")
+
+print(f"[*] Now building unicornafl with maturin")
+run_cmd("maturin develop --release", unicornafl_path, True)
+print(f"[*] Python bindings built, now testing...")
```

Comment:
I think we were using `[+]` for this, instead of `[*]`, or the other way around :D